### PR TITLE
feat: support constraints on string (minLength, maxLength, pattern)

### PR DESCRIPTION
**Pull request by faroceann**
Support JSON schema string contraints:
- minLength
- maxLength
- pattern

Gives more power to downstream validators.

### CODE REVIEW

File: internal/converter/types.go [hunk 1]
```diff
@@ -321,6 +349,7 @@ func (c *Converter) convertField(curPkg *ProtoPackage, desc *descriptor.FieldDes
 		}
 	}
 
+	// Check for custom options
```
Comment (faroceann) [inline internal/converter/types.go:352]:
```suggestion
```

**Comment by chrusty** [comment]
Thank you @faroceann, you've done a thorough job with this PR!

Let me have a closer look then I'll merge it in and make a new release.

And thanks for your interest in the project!

**Review by chrusty — APPROVED** [review]
Hello @faroceann, thank you for this contribution - you've done a thorough job on this PR!

**Comment by chrusty** [comment]
See https://github.com/chrusty/protoc-gen-jsonschema/releases/tag/1.3.7

**Pull request merged**